src/dawn/Optimizer/StencilInstantiation.cpp

```diff
@@ -630,20 +630,19 @@ void StencilInstantiation::setAccessIDNamePairOfGlobalVariable(int AccessID,
   GlobalVariableAccessIDSet_.insert(AccessID);
 }
 
-void StencilInstantiation::removeAccessID(int AccesssID) {
-  if(NameToAccessIDMap_.count(AccessIDToNameMap_[AccesssID]))
-    NameToAccessIDMap_.erase(AccessIDToNameMap_[AccesssID]);
-
-  AccessIDToNameMap_.erase(AccesssID);
-  FieldAccessIDSet_.erase(AccesssID);
-  TemporaryFieldAccessIDSet_.erase(AccesssID);
-
-  auto fieldVersionIt = FieldVersionsMap_.find(AccesssID);
-  if(fieldVersionIt != FieldVersionsMap_.end()) {
-    std::vector<int>& versions = *fieldVersionIt->second;
-    versions.erase(
-        std::remove_if(versions.begin(), versions.end(), [&](int AID) { return AID == AccesssID; }),
-        versions.end());
+void StencilInstantiation::removeAccessID(int AccessID) {
+  if(NameToAccessIDMap_.count(AccessIDToNameMap_[AccessID]))
+    NameToAccessIDMap_.erase(AccessIDToNameMap_[AccessID]);
+
+  AccessIDToNameMap_.erase(AccessID);
+  FieldAccessIDSet_.erase(AccessID);
+  TemporaryFieldAccessIDSet_.erase(AccessID);
+
+  if(variableVersions_.hasVariableMultipleVersions(AccessID)) {
```

Comment:
This check in unnecessary, right?

Reply:
I think it is, because the getVertions is calling a map.at(key) which will throw if the key does not exist